enhancements/cloud-integration/tokenized-auth-enablement-operators-on-cloud.md

```diff
@@ -0,0 +1,340 @@
+the operator to assume the role and authenticate with the Cloud Provider.
+
+In Manual mode or more generally on clusters that support TAT (CCO not necessarily installed), an admin installing an 
+operator is required to:
+
+Extract the CredentialsRequest from the operator's image or codebase in order to know what IAM role is appropriate for 
+the operator to assume
+if using the [Cloud Credential Operator Utility](https://github.com/openshift/cloud-credential-operator/blob/master/docs/ccoctl.md), `ccoctl`:
+1. Find and download a compatible `ccoctl` binary
+2. Accept that `ccoctl` is a binary you downloaded that has god-mode access to your Cloud Provider
+3. Use `ccoctl` to create the IAM Role
+4. Use `ccoctl` to create the secret with the credentials expected by the operator in mint mode
+
+Here is a diagram for how this works on AWS:
+
+![](STS_today.jpg)
+
+This enhancement seeks to unify this process across operators so users of several of them have the same experience and
+similar steps to perform. We aim to automate and reduce those steps as much as possible to make installing operators
+(via OLM) simpler on clusters where TAT authentication is supported.
+
+### User Stories
+
+* As a cluster admin, I want to know which OLM Operators are safe to install because they will not be interacting with
+  the Cloud Provider on a cluster that only supports TAT authentication with the Cloud Provider
+* As a cluster admin, I want to know which OLM Operators support tokenized authentication for my cloud, so that I can
+  provide token-based access for cloud resources for them.
+* As a cluster admin of a cluster using tokenized cloud auth, I want to know what's required to install and upgrade OLM 
+ Operators whenever those operators manage resources that authenticate against my cloud so they can function properly
+* As a cluster admin, I want the experience of TAT authentication to be as similar as possible from one Cloud Provider 
+  to the other, so that I can minimize Cloud Specific knowledge and focus more on OpenShift.
+* As an Operator developer, I want to have a standard framework to define tokenized authentication requirements and
+  consume them, per supported cloud, so that my operator will work on token-based authentication clusters.
+* As an Operator Hub browser, I want to know which operators support tokenized cloud auth and on which clouds so I can
+  see only a filtered list of operators that will work on the given cluster.
+* As an Operator Hub browser, I want to be informed / reminded in the UI that the cluster only supports TAT 
+  authentication with the Cloud Provider, so that I don't confuse the cluster with one that will try to mint long lived
+  credentials.
+* As an Operator Hub browser, I want to be able to easily provide what's required to the OLM operators I install through
+  the UI.
+* As the HyperShift team, where CCO is not installed so the only supported authentication mode is via TAT, I want the
```

Comment:
so if the operators work in tat mode, cco is not required? if so, i think it's not written clearly in the enhancement - i reached this part without understanding it

Reply:
We're proposing to add CCO with this new TAT detection code, running in Manual mode, back into HyperShift. I'll try to make that more clear here.